spec/runners/unified/assertions.rb

```diff
@@ -51,6 +51,31 @@ def assert_result_matches(actual, expected)
             end
           end
 
+          %w(acknowledged).each do |k|
```

Comment:
I decided to move all comparisons with result object to this method. This is not necessarily applicable to dots and dollars.